include/cetl/unbounded_variant.hpp

```diff
@@ -556,27 +557,55 @@ struct base_access : base_storage<Pmr, Footprint, Alignment>
     template <typename Tp, std::enable_if_t<is_rtti_convertible<Tp>, int> = 0>
     void make_converters() noexcept
     {
-        value_const_converter_ = [](const void* const storage, const type_id& id) {
-            const auto ptr = static_cast<const Tp*>(storage);
-            return ptr->_cast_(id);
+        value_const_converter_ = [](const void* const storage, const cetl::type_id& dst_type_id) {
```

Comment:
These lambdas may fail on systems with no heap and a deficient small-value-optimization in std::function. Please use function pointers to avoid this.

Reply:
I don't think so, they are just pure (implicitly generated) static functions, they don't capture anything (neither any of outside vars nor `this` - so no allocation. sizeof of them is 1 (should kinda be zero but as you know even empty struct sizeof is 1). And the fact that I can just assign them to a simple pointer to function (see `value_const_converter_` declaration below) proves it - as soon as add anything to the capture list such assignments won't work anymore.

Here is sizeof of this lambda:
```
        using Xxx = decltype([](const void* const storage, const cetl::type_id& dst_type_id) {
            CETL_DEBUG_ASSERT(nullptr != storage, "");
            const auto        ptr     = static_cast<const Tp*>(storage);
            const void* const dst_ptr = ptr->_cast_(dst_type_id);
            return std::make_pair(dst_ptr, cetl::type_id_value<Tp>);
        });
        static_assert(sizeof(Xxx) == 1);
```

Reply:
In addition, although I could technically make explicit template static member functions for convertors, I'm reluctant todo so b/c:
1. There will be 4 of them (see two mutually exclusive versions of `make_converters`).
2. There all will be out of `make_converters` context
3. `make_converters` is the only supposed entity to reference them; and of course nobody is allowed to call them directly (only indirectly via `value_[mut|const]_convertor_` function pointers). So, in some sense I believe am doing proper isolation here without introducing any indeterministic behavior.

Reply:
Furthermore, I scanned AUTOSAR14 spec, and found no lambda rule against my implementation. The only remotely applicable rule is the following:
>Rule A5-1-6 (advisory, implementation, automated)
Return type of a non-void return type lambda expression should be
explicitly specified.

I can fix it.

Reply:
done - see [commit](https://github.com/OpenCyphal/CETL/pull/127/commits/e0adc23126e5c014cdd295c5295d3bf5887fa4ce)

Reply:
To back up what Sergei said:

<img width="655" alt="image" src="https://github.com/OpenCyphal/CETL/assets/3298404/d129c906-c173-4a9d-8000-1c926ce36dd8">